cli/src/commands/migration/status.ts

```diff
@@ -0,0 +1,81 @@
+import { Args } from '@oclif/core';
+import { BaseCommand } from '../../base.js';
+import { getBranchDetailsWithPgRoll, isBranchPgRollEnabled } from '../../migrations/pgroll.js';
+import chalk from 'chalk';
+import { match } from 'ts-pattern';
+
+export default class MigrationStatus extends BaseCommand<typeof MigrationStatus> {
+  static description = 'Get the status of the last pgroll migration';
+
+  static examples = [];
+
+  static flags = {
+    ...this.commonFlags,
+    ...this.databaseURLFlag
+  };
+
+  static args = {
+    branch: Args.string({ description: 'The branch to fetch the status for', required: true })
+  };
+
+  static hidden = true;
+
+  async run() {
+    const { args, flags } = await this.parseCommand();
+
+    const xata = await this.getXataClient();
+    const { workspace, region, database, branch } = await this.getParsedDatabaseURLWithBranch(
+      flags.db,
+      args.branch,
+      true
+    );
+
+    const details = await getBranchDetailsWithPgRoll(xata, { workspace, region, database, branch });
+
+    if (!isBranchPgRollEnabled(details)) {
+      this.error(`${chalk.gray('xata migration')} commands are only supported in Postgres enabled databases`);
+    }
+
+    const commonParams = {
+      region,
+      workspace,
+      dbBranchName: `${database}:${branch}`
+    };
+
+    const migrationJobStatus = await xata.api.migrations.getBranchMigrationJobStatus({
+      pathParams: {
+        ...commonParams
+      }
+    });
+
+    const statusChalkColor = match(migrationJobStatus.status)
```

Comment:
I like `ts-pattern` as it forces you to explicitly handle all states. Let me know if there are any reservations against introducing it to this repo.